fe/src/main/java/org/apache/doris/clone/TabletChecker.java

```diff
@@ -0,0 +1,213 @@
+
+import org.apache.doris.catalog.Catalog;
+import org.apache.doris.catalog.Database;
+import org.apache.doris.catalog.MaterializedIndex;
+import org.apache.doris.catalog.OlapTable;
+import org.apache.doris.catalog.Partition;
+import org.apache.doris.catalog.Table;
+import org.apache.doris.catalog.Tablet;
+import org.apache.doris.catalog.Tablet.TabletStatus;
+import org.apache.doris.common.Pair;
+import org.apache.doris.common.util.Daemon;
+import org.apache.doris.system.SystemInfoService;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Sets;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.Set;
+
+/*
+ * This checker is responsible for checking all unhealthy tablets.
+ * It does not responsible for any scheduler of tablet repairing or balance
+ */
+public class TabletChecker extends Daemon {
+    private static final Logger LOG = LogManager.getLogger(TabletChecker.class);
+
+    private static final long INTERVAL_MS = 60 * 1000L; // 1min
+
+    private Catalog catalog;
+    private SystemInfoService infoService;
+    private TabletScheduler tabletScheduler;
+    private TabletSchedulerStat stat;
+
+    // db id -> (tbl id -> partition id)
+    // priority of replicas of partitions in this table will be set to VERY_HIGH if not healthy
+    private com.google.common.collect.Table<Long, Long, Set<Long>> prios = HashBasedTable.create();
+
+    public TabletChecker(Catalog catalog, SystemInfoService infoService, TabletScheduler tabletScheduler,
+            TabletSchedulerStat stat) {
+        super("tablet checker", INTERVAL_MS);
+        this.catalog = catalog;
+        this.infoService = infoService;
+        this.tabletScheduler = tabletScheduler;
+        this.stat = stat;
+    }
+
+    public void addPrios(long dbId, long tblId, List<Long> partitionIds) {
+        Preconditions.checkArgument(!partitionIds.isEmpty());
+        synchronized (prios) {
+            Set<Long> parts = prios.get(dbId, tblId);
+            if (parts == null) {
+                parts = Sets.newHashSet();
+                prios.put(dbId, tblId, parts);
+            }
+            parts.addAll(partitionIds);
+        }
+
+        // we also need to change the priority of tablets which are already in
+        tabletScheduler.changePriorityOfTablets(dbId, tblId, partitionIds);
+    }
+
+    /*
+     * For each cycle, TabletChecker will check all OlapTable's tablet.
+     * If a tablet is not healthy, a TabletInfo will be created and sent to TabletScheduler for repairing.
+     */
+    @Override
+    protected void runOneCycle() {
+        boolean schedulerHasTask = !tabletScheduler.isEmpty();
+
+        // if scheduler has tasks, we only check tablets in prios
+        checkTablets(schedulerHasTask);
+        stat.counterTabletCheckRound.incrementAndGet();
+
+        LOG.info(stat.incrementalBrief());
```

Comment:
The exception of `checkTablets` need catching in `runOneCycle`

Reply:
Here we only catch expected exception, any unexpected exception is caught in `Deamon`